src/cider/nrepl/middleware/inspect.clj

```diff
@@ -48,47 +53,79 @@
 
 (defn pop-reply
   [{:keys [transport] :as msg}]
-  (let [inspector (swap-inspector! msg inspect/up)]
-    (transport/send
-     transport
-     (response-for msg :value (:rendered inspector) :status :done))))
+  (try
+    (let [inspector (swap-inspector! msg inspect/up)]
+      (transport/send
+       transport
+       (response-for msg :value (:rendered inspector) :status :done)))
+    (catch Exception e
+      (transport/send
+       transport
+       (response-for msg (u/err-info e :inspect-pop-error))))))
 
 (defn push-reply
   [{:keys [idx transport] :as msg}]
-  (let [idx (Integer/parseInt idx)
-        inspector (swap-inspector! msg inspect/down idx)]
-    (transport/send
-     transport
-     (response-for msg :value (:rendered inspector) :status :done))))
+  (try
+    (let [idx (parse-int idx)
```

Comment:
As mentioned in the PR - it doesn't make sense to expect a string here. Let's just expect a number and change the related elisp code as well.